tests/test_authentication.py

```diff
@@ -94,6 +105,14 @@ def test_unknown_creds(self):
 
         eq_(exc.exception.detail, 'authentication failed')
 
+    def test_alternative_credential_lookup(self):
+        sender = self._sender(credentials=ALTERNATIVE_CREDS)
+        req = self._request(sender)
+        lookup_path="tests.test_authentication.alternative_lookup"
```

Comment:
For a little bit of flexibility in case this file moves around it would be better to reference this path by module name. You can do it like `lookup_path = '%s.alternative_lookup' % __name__`. I'm *pretty sure* that will work.